content/api/commands/intercept.md

```diff
@@ -212,9 +214,41 @@ For more guidance around aliasing requests with GraphQL, see [Working with Graph
 
 </Alert>
 
+### Matching `url` and `method` with [RouteMatcher](#routeMatcher-RouteMatcher)
+
+Specifying a `method` and `url` to match can also be acheived by passing the `routeMatcher` object into `cy.intercept` instead:
+
+```js
+// These both yield the same result:
+cy.intercept({ method: 'GET', url: '**/users' })
+cy.intercept('GET', '**/users')
+```
+
+### Pattern Matching
+
+```js
+// match updates to the `/users` endpoint
+cy.intercept({
+  method: '+(PUT|PATCH)',
```

Comment:
I see a mention in the the latter example `method: '/PUT|PATCH/'` that it's a regex, but I don't see an explanation here of what format `method: '+(PUT|PATCH)'` is.

Additionally, in reading Arguments -> method, I don't either of these usages explained. We'll need to document this there.

Reply:
Superb. I updated the comment in the snippet and the types in the heading for `method`.

Reply:
> Superb. I updated the comment in the snippet and the types in the heading for `method`.

If you push up a commit with that change, I'll review it. No need to wait until everything is done to push.

Reply:
In the heading for `method` I see that you added `Glob` and `RegExp` but I see a few problems with this.

1. `Glob` is not actually a type. It's just a way to format a string. I do see [GlobPattern](https://github.com/cypress-io/cypress/blob/7192a78ef273a0d0a0811721cf69a88da54cf276/packages/net-stubbing/lib/external-types.ts#L205) but that doesn't really explain to an end user what a glob is. Instead of listing `Glob` as a type in the header, mention/explain it in the description.
2. `RegExp` is a type, but the only `method` examples I see use `String` values. Is `RegExp` actually supported? eg. `cy.intercept(/PUT/PATCH/, '/my/url')`? If it is, we may want to show an example. If not, we should mention/explain the regex-as-string format in the description.

Reply:
I'm being consistent with the rest of the docs. If you want to change it everywhere, create an issue.

Reply:
These are great thoughts but let's get this in and address it in a future issue.

Reply:
I'm trying to understand which docs this is consistent with. The [`cy.route` Arguments](https://docs.cypress.io/api/commands/route#Arguments) `match` only mentions `String` as a type (although `url` mentions `Glob` and `RegExp`, which makes sense to me).

For consistency's sake, how about we change `method` to exactly match what `cy.route` has, with the one difference being, in the callout, saying "all" instead of "GET"?

Reply:
It would be really helpful if you made code suggestions.

Reply:
Sure, let me give that a try! I'm still not used to that feature.